tests/space-time-stack/CMakeLists.txt

```diff
@@ -1,18 +1,10 @@
 # A function to add such "simple" tests in 'tests/CMakeLists.txt' might be a good option.
-add_executable(test_space_time_stack_demangling)
-target_sources(
-    test_space_time_stack_demangling
-    PRIVATE
-        test_demangling.cpp
+add_one_test(
+    TARGET_NAME test_space_time_stack_demangling
+    SOURCE_FILE test_demangling
 )
-target_link_libraries(
+target_compile_definitions(
```

Comment:
Why are you using a compile definition for this?
It does not look like the right tool for the job.
Why not use a command line argument or an environment variable?

Reply:
For now, the test looks like this:
```c++
TEST(SpaceTimeStackTest, demangling) {
  //! Fake arguments a user would give via CLI.
  std::string arg{"--kokkos-tools-libs=" LIBKP_SPACE_TIME_STACK};
  int argc = 1;
  std::vector<char*> argv{arg.data(), nullptr};

  //! Initialize @c Kokkos.
  Kokkos::initialize(argc, argv.data());
```
and so it's true that the path to the library is carved in stone. I would argue that it's no big deal since (my guess is) no one will ever use this test with another library :angel:

But I'm OK with changing the approach to something more flexible :wink:

Are you suggesting that I should not hardcode the library path in the executable, and rather read it from the environment ? Something like what follows:
```c++
TEST(SpaceTimeStackTest, demangling) {
  //! Fake arguments a user would give via CLI.
  std::string arg{"--kokkos-tools-libs=" + getenv("LIBKP_SPACE_TIME_STACK")};
  int argc = 1;
  std::vector<char*> argv{arg.data(), nullptr};

  //! Initialize @c Kokkos.
  Kokkos::initialize(argc, argv.data());
```
with the test having additional environment like:
```cmake
set_property(
        TEST "..."
        APPEND
        PROPERTY
            ENVIRONMENT "LIBKP_SPACE_TIME_STACK=..."
    )
```

Reply:
Kokkos looks for `KOKKOS_TOOLS_LIBS`.  We don't need to invent a new way to enable.

Reply:
Indeed!

I've changed `kp_add_executable_and_test` such that it also takes the `CONNECTOR_LIB`. The test environment is enriched with `"KOKKOS_TOOLS_LIBS=$<TARGET_FILE:${kaeat_args_CONNECTOR_LIB}>"`.

Is this approach what you had in mind ?